avocado/core/jobdata.py

```diff
@@ -122,16 +121,6 @@ def get_variants_path(resultsdir):
     return _retrieve(resultsdir, VARIANTS_FILENAME)
 
 
-def retrieve_variants(resultsdir):
-    """
-    Retrieves the job variants object from the results directory.
-    """
-    recorded_variants = _retrieve(resultsdir, VARIANTS_FILENAME)
-    if recorded_variants:
-        with open(recorded_variants, 'r') as variants_file:
-            return varianter.Varianter(state=json.load(variants_file))
```

Comment:
I like where you're taking this... but I have a (large) number of additional changes to the varianter that will probably conflict here.  I'll try to send the code so that it's "explained" there.

Reply:
OK, so there's no conflict in my first batch of work regarding the varianter, so I'm picking this patch of yours in adding to my PR #4042 (I mean to help so you can carry less patches not directly related to the multiple test suites work).

Reply:
Yes, no problem. Feel free to do this.